test/Concurrency/Runtime/async_taskgroup_addUnlessCancelled.swift

```diff
@@ -0,0 +1,64 @@
+// RUN: %target-run-simple-swift( -target %target-swift-5.1-abi-triple -parse-as-library) | %FileCheck %s
+
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+
+// rdar://76038845
+// REQUIRES: concurrency_runtime
+// UNSUPPORTED: back_deployment_runtime
+
+import Dispatch
+
+@available(SwiftStdlib 6.0, *)
+func test_withTaskGroup_addUnlessCancelled() async throws {
+  let task = Task {
+    await withTaskGroup(of: Void.self) { group in
+      print("Inner: Sleep...")
+      try? await Task.sleep(nanoseconds: 2_000_000_000)
```

Comment:
Could we somehow use continuations rather than `Task.sleep()`? I appreciate that we're waiting 2s here and only 1ms on line 27 below, so it's _unlikely_ to spuriously fail, but it still isn't impossible.

Reply:
Yeah I can rework this a bit. Tbh "either timing" we want to test so maybe I'll just drop the sleep entirely